CMakeLists.txt

```diff
@@ -1,6 +1,19 @@
 cmake_minimum_required(VERSION 3.18)
 project(Arduino-Filters CXX)
-set(CMAKE_CXX_STANDARD 14)
+
+# GoogleTest requires at least C++17
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+include(FetchContent)
+FetchContent_Declare(
+  googletest
+  URL https://github.com/google/googletest/archive/03597a01ee50ed33e9dfd640b249b4be3799d395.zip
```

Comment:
- From your PR comment *GTest is only downloaded if the root CMakeLists.txt is used (for testing ArduinoFilters)*, I assume this just automates downloading a dependency, but does not change any functionality? 
- Do you think it's worth adding a comment here saying why googletest is needed?